### PR TITLE
 [CELEBORN-158][Flink] Add ShuffleServiceFactory to Support MapPartition in …

### PR DESCRIPTION
**Pull request by zhongqiangczq**
…flink-plugin

### What changes were proposed in this pull request?

Currently, flink plugin lacks the code that shuffleServiceFactory creates shuffleWrites.
so this pr completes the code for flink plugin 

### Does this PR introduce _any_ user-facing change?

@waitinfuture @FMX @RexXiong

### CODE REVIEW

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleServiceFactory.java [hunk 1]
```diff
@@ -35,6 +53,41 @@ public ShuffleMaster<RemoteShuffleDescriptor> createShuffleMaster(
   public ShuffleEnvironment<ResultPartitionWriter, IndexedInputGate> createShuffleEnvironment(
       ShuffleEnvironmentContext shuffleEnvironmentContext) {
     // TODO
```
Comment (RexXiong) [inline client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleServiceFactory.java:55]: delete this line

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 2]
```diff
@@ -2772,4 +2773,13 @@ object CelebornConf extends Logging {
       .doc("The time before a cache item is removed.")
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("15s")
+
+  val MEMORY_PER_RESULT_PARTITION: ConfigEntry[String] =
+    buildConf("celeborn.network.memory.per.result.partition")
```
Comment (RexXiong) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2778]: use client prefix for this configuration

File: client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleServiceFactory.java (hunk 1 above)
Comment (zhongqiangczq) [reply client-flink/flink-1.14/src/main/java/org/apache/celeborn/plugin/flink/RemoteShuffleServiceFactory.java]: has deleted

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala (hunk 2 above)
Comment (zhongqiangczq) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: modified to celeborn.client.network

**Review by RexXiong — APPROVED** [review]
LGTM, Thanks! @waitinfuture

**Review by waitinfuture — APPROVED** [review]

**Pull request merged**